apache/metron
# METRON-1654: findOne request after an alert patch returns with the original state of the alert item

## Contributor Comments
With the addition of https://github.com/apache/metron/pull/1190, the Alerts UI can now update state using the object returned from a REST update call.  This PR updates the Alerts UI to do that.

### Testing
This has been tested in full dev.  

1. Spin up full dev and navigate to the Alerts UI
2. Open an alert and change the status from "NEW" to "OPEN".  After closing the detail panel the status for that alert should show as "OPEN" in the list view.
3. Create a metaalert and search for the metaalert so that is shows up in the list view.  Add an alert to that metaalert by checking the checkbox next to an alert and using the "Add to Alert" option in the "Actions" dropdown.  The count next the the metaalert in the list view should increment by 1.
4. Remove an alert from the metaalert.  The count next to the metaalert should decrement by 1 and the alert should be removed immediately.
5. Other update functions in the UI should be tested to ensure there are no regressions:  adding/deleting a comment, changing a metaalert name, etc.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
